src/check-c-linkage-decls.py

```diff
@@ -4,9 +4,9 @@
 
 os.chdir (os.environ.get ('srcdir', os.path.dirname (__file__)))
 
-HBHEADERS = os.environ.get ('HBHEADERS', '').split () or \
+HBHEADERS = [os.path.basename (x) for x in os.environ.get ('HBHEADERS', '').split ()] or \
```

Comment:
Why these are needed? File objects in meson (created using by `files()` in this patch) when turned to string are like `src/hb-x.hh` instead just the base name they used to, alternative approach would be top drop the above `chdir` yet I like to keep the script still runnable from command line that's why I am doing this for now.